src/components/05_pages/NodeForm/index.js

```diff
@@ -132,25 +109,76 @@ class NodeForm extends React.Component {
     } else {
       fieldData = Object.values(data);
     }
-    this.setState(prevState => ({
-      entity: {
-        data: {
-          ...prevState.entity.data,
-          relationships: {
-            ...prevState.entity.data.relationships,
-            [fieldName]: {
-              data: fieldData,
+    this.setState(
+      prevState => ({
+        entity: {
+          data: {
+            ...prevState.entity.data,
+            relationships: {
+              ...prevState.entity.data.relationships,
+              [fieldName]: {
+                data: fieldData,
+              },
             },
           },
         },
-      },
-    }));
+      }),
+      () => this.props.onChange(this.props.bundle, this.state.entity),
+    );
   };
 
   getSchemaInfo = (schema, fieldName) =>
     schema.properties.data.properties.attributes.properties[fieldName] ||
     schema.properties.data.properties.relationships.properties[fieldName];
 
+  calculateState = (prevProps, prevState, setState) => {
```

Comment:
I'm wondering if there would be a better name for this function; maybe `refreshState` would be more accurate?

Reply:
Or `rebuildState`?

Reply:
How about ```deriveState```?